src/Controller/FightController.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace App\Controller;
+
+use App\Model\CastleManager;
+use App\Model\SessionManager;
+use App\Model\TroopManager;
+
+class FightController extends AbstractController
+{
+    /**
+     * recovery of game info (score, round, trooper's player)
+     * create ennemy troops
+     * storage before fight resolution in session
+     * return to the get_ready view
+    */
+    public function getTroops(): string
+    {
+        session_start();
+        // recovery of game info (score, round)
+        $gameInfo = new CastleManager();
+        $game = $gameInfo->selectGame();
+        // recovery of player troop and creation of enemy troop
+        $troopManager = new TroopManager();
+        $troops = $troopManager->selectTroopPlayer();
+        $enemy = $troopManager->selectTroopEnemy((rand(1, 3)), 5);
+        // to stall troops in a session
+        $session = new SessionManager();
+        $_SESSION = $session->saveElement($enemy, $troops, $game);
+        //return to view get_ready
+        return $this->twig->render('Fight/get_ready.html.twig', [
+            'troops' => $troops,
+            'session' => $_SESSION,
+            'game' => $game,
+        ]);
+    }
+    /**
+     * save castle data
+     */
+    private function saveCastle($newScore, $round): void
+    {
+        $save = new CastleManager();
+        $save->setRound($round);
+        $save->setScore($newScore);
+    }
+    /**
+     * Update and save vigor data
+     */
+    private function saveVigor($id): void
+    {
+        $updateVigor = new TroopManager();
+        $updateVigor->updateVigor($id);
+    }
```

Comment:
If you had the Troop object update it's vigor itself when the fight method is called, you'll only need to use the troopManager to update the database after a fight.
You could have a more simple code